pkg/authenticator/authenticator.go

```diff
@@ -144,6 +144,10 @@ func (auth *Authenticator) Login() error {
 		nil,
 	)
 	if err != nil {
+		injectClientCertError := getInjectClientCertError(auth.Config.InjectCertLogPath)
```

Comment:
So most of my other comments are small nits but it seems a bit strange that the client is in charge of reporting its own injection error as it can only cover a small percentage of issues arising from the injection failures. I think a more robust solution would be that the server parses this file on errors over the same APIs that it used to inject the cert or maybe doing something even more novel.

Reply:
the issue is that the server will not get these errors as it closes the socket without waiting for a reply from the other side (k8s api). You can read more about it [here](https://github.com/cyberark/conjur/blob/master/design/authenticators/authn_k8s/inject_client_cert_solution_design.md#waiting-for-the-injection-to-finish) (and the rest of the solution design may be interesting as well).

Thus, the only way that the customer will see the error is in the log of the client. It will enhance supportability as the client logs will show that it failed and why. Ideally we would have this log in the server but we chose not to wait for the request to end to enhance the request's performance.

Reply:
Ah ok - I was not aware that the current async flow will be kept.